app/models/navigation_type.rb

```diff
@@ -0,0 +1,27 @@
+class NavigationType
+  def initialize(content_item)
+    @content_item = content_item
+  end
+
+  def should_present_taxonomy_navigation?
```

Comment:
Possible alternative way of writing this (although it's obviously a matter of personal preference):

```ruby
def should_present_taxonomy_navigation?
  return true if tagged_to_world_wide_taxonomy?
  return false if content_is_tagged_to_browse_pages?
  
  content_is_tagged_to_a_taxon?
end
```